virtcontainers/sandbox.go

```diff
@@ -946,6 +946,28 @@ func (s *Sandbox) DeleteContainer(containerID string) (VCContainer, error) {
 	return c, nil
 }
 
+// StatusContainer gets the status of a container
+func (s *Sandbox) StatusContainer(containerID string) (ContainerStatus, error) {
```

Comment:
This function should be smarter than this. It should be inspired from what is done in `virtcontainers/api.go`:
```go
func statusContainer(sandbox *Sandbox, containerID string) (ContainerStatus, error) {
```
The reason is that we need the status to be updated if virtcontainers realizes the shim has terminated. Otherwise, this assumes the caller will monitor the shim process in order to call `StopContainer()` when the shim terminates, but I don't think we want to rely on that.

Reply:
@sboeuf I'm afraid the shim pid is not enough for representing container status because there can be no shim at all in the kata_builtin_shim case, which is the main usage of frakti along with this API. I'd like to handle it properly in a separate PR since it's an existing issue `StatusContainer()` as well in api.go. WDYT?

Reply:
@bergwolf fair enough since this won't break the existing API. In the meantime (until you submit this follow up PR), could you add some warning comments on top of this function to explain that it should not be used as it might return the wrong state.

Reply:
@sboeuf how about tracking it in a separate issue (kata-containers/runtime/issues/253) instead? An outstanding issue works better at tracking than comments in the code. Besides, currently the CI system is broken. I don't want to wait another day just for CI to pass. I'm having other PRs piling up on top of this one.

Reply:
updated. Thanks @sboeuf !